trieste/models/model_interfaces.py

```diff
@@ -341,15 +345,26 @@ class GaussianProcessRegression(GPflowPredictor, TrainableProbabilisticModel):
     or :class:`~gpflow.models.SGPR`.
     """
 
-    def __init__(self, model: GPR | SGPR, optimizer: Optimizer | None = None):
+    def __init__(
+        self, model: GPR | SGPR, optimizer: Optimizer | None = None, num_kernel_samples: int = 10
+    ):
         """
         :param model: The GPflow model to wrap.
         :param optimizer: The optimizer with which to train the model. Defaults to
             :class:`~trieste.models.optimizer.Optimizer` with :class:`~gpflow.optimizers.Scipy`.
+        :param num_kernel_samples: Number of randomly sampled kernels (for each kernel parameter) to
+            evaluate before beginning model optimization. Therefore, for a kernel with `p`
+            parameters, we evaluate `p * num_kernel_samples` kernels.
```

Comment:
maybe this is slightly confusing? change to "for a kernel with p (vector-valued) parameters"